mitodl/micromasters
# Add contributors when new channel is created

#### What are the relevant tickets?
Part of #3475

#### What's this PR do?
Adds contributors when a new channel is created, and synchronizes new contributors when the account is created or updated

#### How should this be manually tested?
Creating a channel should populate contributors based on the provided query:
 - Seed your database using seed_db
 - Go to `/learners`, open your network tab and execute some search targeting a handful of people with some substring of their name, for example `fl`. Copy the search query from the request.
 - Go to `/api/v0/channels/` and fill out the form. Paste the query into the search query field. You should get a 201 with information about the new channel
 - On open-discussions go to `/api/v0/channels/<channel_name>/contributors/` and verify that there are the right number of users (all the users in your search plus one for the API user)

When a user updates their profile it should unenroll or enroll them via profile signal:
 - Pick a user who matched the search criteria earlier. Log in as that user on reddit. You can change the password like this:
    
        from r2.models.account import *
        a = Account._by_name("your_username_here")
        a.password = bcrypt_password("your new password")
        a._commit()

You will also need to edit the max length for the input field on the login page to allow the ULID

 - Update the profile so that their name does not fit the search criteria anymore. Verify that the user is now not a contributor and they do not see that channel in their list of subscriptions.
  - Update the profile so that their name is now in the search criteria again. Verify that the user is a member of the channel again.
